presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java

```diff
@@ -395,4 +430,271 @@ private static List<Property> applyPropertyOverride(List<Property> properties, L
                 .map(entry -> new Property(new Identifier(entry.getKey()), entry.getValue()))
                 .collect(toImmutableList());
     }
+
+    private static class NonDeterministicColumnRewriter
+            extends AstVisitor<Node, Void>
+    {
+        private static class FunctionCallSubstitute
+        {
+            private final QualifiedName name;
+            private final List<Integer> originalArgumentIndices;
+
+            private FunctionCallSubstitute(QualifiedName name, List<Integer> originalArgumentIndices)
+            {
+                this.name = requireNonNull(name, "name is null");
+                this.originalArgumentIndices = ImmutableList.copyOf(requireNonNull(originalArgumentIndices, "originalArgumentIndices is null"));
+            }
+        }
+
+        private static final Map<QualifiedName, FunctionCallSubstitute> FUNCTION_CALL_SUBSTITUTE_LOOKUP = ImmutableMap.<QualifiedName, FunctionCallSubstitute>builder()
+                .put(QualifiedName.of("approx_percentile"), new FunctionCallSubstitute(QualifiedName.of("avg"), ImmutableList.of(1)))
+                .put(QualifiedName.of("approx_distinct"), new FunctionCallSubstitute(QualifiedName.of("count"), ImmutableList.of(1)))
```

Comment:
Wondering what is the use case we are trying to solve for here.
These functions are already skipped in the sampling [pipeline](https://www.internalfb.com/code/fbsource/fbcode/dataswarm-pipelines/tasks/di/presto/sampling.py?lines=188-190).

Reply:
Hi @ajaygeorge , the use case here is to include these queries, as opposed to filtering them out, so that we can verify the correctness of other columns or at the query level. All of one type of Deltoid queries contain approx_percentile() UDF columns, and we need to cover the type by verifier.

Reply:
@gggrace14  Thanks. If that is the use case, you will have to also build out a new sampling pipeline for generating release/nightly suites for deltoid queries since this code change will not help since the filtering is already done in the sampling stage.